### PR TITLE
CORE-6250 Context Propagation to External processors

### PR DESCRIPTION
**Pull request by simon-johnson-r3**
API PR here: https://github.com/corda/corda-api/pull/559

* Flow Context is added as a single flat map to external event context.
* Not in scope of this story how those processors use the context. As little is populated right now, this is more about plumbing it in that using it usefully.
* Most of the PR is test churn.
* Much of this code has to execute within a fiber and cannot be tested easily. This seems to have been the same in the ExternalEvent story too. To test I added a log line into the DB worker (not in this PR), this is the output:
```
corda.log:2022-09-08 13:30:20.469 [durable processing thread virtual.node.entity.processor-db.entity.processor] ERROR net.corda.entityprocessor.impl.internal.EntityMessageProcessor - @@@ external event contextProperties:'{"requestId": "ed2eb212-53a8-44a7-908a-045f315c328e", "flowId": "1007e8f3-c315-4626-9214-745188ccf269", "contextProperties": {"items": [{"key": "corda.account", "value": "account-zero"}]}}'
```
Here you can see the stock account zero context, which was added all the way at the RPC level, make it to the external processors.

### CODE REVIEW

**Review by mnesbit — DISMISSED** [review]
OK

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/external/events/impl/executor/ExternalEventExecutorImpl.kt [hunk 1]
```diff
@@ -25,16 +26,27 @@ class ExternalEventExecutorImpl @Activate constructor(
...
+                        parameters,
+                        externalContext(this)
+                    )
                 )
-            )
+            }
         )
     }
 
+    private fun externalContext(flowFiber: FlowFiber) =
```
Comment (lankydan) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/external/events/impl/executor/ExternalEventExecutorImpl.kt]: Without the return type its a bit hard to know what this is returning

**Review by lankydan — DISMISSED** [review]
One comment, but generally fine.

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/external/events/impl/executor/ExternalEventExecutorImpl.kt (hunk 1 above)
Comment (simon-johnson-r3) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/external/events/impl/executor/ExternalEventExecutorImpl.kt]: I've made the change. Given it's non functional and I've already bumped the version ready for the new version of the api, I will submit this at the same time I push the version, once the new api is published.

**Review by lankydan — APPROVED** [review]

**Pull request merged**